### PR TITLE
SplittingInterval cleanup

### PR DESCRIPTION
**Pull request by peterfpeterson**
This does two changes:
* Changes the name of the header that defines `SplittingInterval` to match the class name
* Changes the alias that represents a vector of `SplittingInterval` to more closely represent what it is an alias for. The old name was misleading

**To test:**

This is a rename so there isn't anything to test. The builds should still pass.

*There is no associated issue.*

*This does not require release notes* because it is an internal renaming

---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there is GUI work does it follow the [GUI standards](http://developer.mantidproject.org/Standards/GUIStandards.html)?
- If there are changes in the release notes then do they describe the changes appropriately?
- Are the release notes saved in a separate file, using Issue or PR number for file name and in the correct location?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.

### CODE REVIEW

**Review by jmborr — COMMENTED** [review]
Wow, lot's of weeding.
Everything looks good but I had a question, more than a comment, for two `included` that have been removed.

File: Framework/Algorithms/test/ExportTimeSeriesLogTest.h [hunk 1]
```diff
@@ -22,7 +22,6 @@
 #include "MantidDataObjects/TableWorkspace.h"
 #include "MantidFrameworkTestHelpers/WorkspaceCreationHelper.h"
 #include "MantidKernel/TimeSeriesProperty.h"
-#include "MantidKernel/TimeSplitter.h"
```
[the remark this answers is not in the document]
Comment (peterfpeterson) [reply Framework/Algorithms/test/ExportTimeSeriesLogTest.h:25]: The compiler is more than happy to include things you never use. The other way around doesn't turn out so well. We've had various tries at using [include-what-you-use](https://include-what-you-use.org/) on mantid, but it took too long to run to be practical.

**Review by jmborr — DISMISSED** [review]

File: Framework/Algorithms/test/ExportTimeSeriesLogTest.h (hunk 1 above)
[the remark this answers is not in the document]
Comment (peterfpeterson) [reply Framework/Algorithms/test/ExportTimeSeriesLogTest.h:25]: If you are feeling bold, iwyu and cmake have both improved their inter-operation with each other ([instructions](https://github.com/include-what-you-use/include-what-you-use#using-with-cmake))

**Review by jmborr — DISMISSED** [review]
I didn't find any more instances of `TimeSplitterType` in your branch :smile:

**Review by jmborr — APPROVED** [review]

**Pull request merged**